### PR TITLE
Add anonymous login

### LINKED ISSUE #62: Allow anonymous users
Right now, some people might not want to even test the app if it require logging in before hand.

### PR DESCRIPTION
**Pull request by gbaranski**
closes #62

### CODE REVIEW

**Comment by gbaranski** [comment]
@stasbar done

**Comment by stasbar** [comment]
lemme see

**Comment by gbaranski** [comment]
@stasbar yo eazy how is it going?

**Comment by stasbar** [comment]
w8 a moment, I just need to come off the piano for a moment

File: src/services/user.ts [hunk 1]
```diff
@@ -98,3 +99,19 @@ export async function logoutFirebase(): Promise<void> {
...
 }
+
+export async function logoutFirebaseWithRedirect() {
+  await logoutFirebase();
+  history.replace('/login');
+}
+
+export function notifyToLogIn() {
+  notification.open({
+    message: 'Only for logged users!',
```
Comment (stasbar) [inline src/services/user.ts:110]: You need to be logged in

File: src/services/user.ts [hunk 2]
```diff
@@ -98,3 +99,19 @@ export async function logoutFirebase(): Promise<void> {
...
+    description: 'Click on the notification to redirect to login page',
```
Comment (stasbar) [inline src/services/user.ts:111]: Click to login

Comment (gbaranski) [reply src/services/user.ts]: Done

File: src/services/user.ts (hunk 1 above)
Comment (gbaranski) [reply src/services/user.ts]: Done

**Comment by gbaranski** [comment]
@stasbar done

**Comment by stasbar** [comment]
display "You need to be logged in" instead. Basically, everywhere currentUser is used and is undefined (because is anonymous) should display "You need to be logged in"
Screenshot 2020-08-18 at 21 52 18

**Comment by stasbar** [comment]
Screenshot 2020-08-18 at 21 52 43
Prevent adding new content

**Comment by stasbar** [comment]
@gbaranski have you seen my comments ?

**Comment by gbaranski** [comment]
@stasbar yes i will fix that in a moment

**Comment by gbaranski** [comment]
Screen Shot 2020-08-27 at 23 40 07

working, users are unable to edit anything when they are logged as anonymous

Screen Shot 2020-08-27 at 23 41 06

create button is gone when logged as anonymous

File: src/components/ItemView/ItemView.tsx [hunk 3]
```diff
@@ -47,9 +48,27 @@ export function Actions<T extends Photo | Post | Link | Coil | Liquid>({
...
+    }
+    report(what, item.uid, user.uid);
+  };
+  const submitComment = () => {
+    if (currentUser.isAnonymous) {
+      message.error('You need to be logged in');
+      return;
+    }
+    commentItem(what, draftComment, item.uid, user);
+  };
```
Comment (stasbar) [inline src/components/ItemView/ItemView.tsx:71]: create `usersOnly` middleware so the common logic can be extracted.

```
const onLikeClick = usersOnly(() => {
    like(what, item.uid, user.uid);
})
const onReportClick = usersOnly(() => {
    report(what, item.uid, user.uid);
})
const submitComment = usersOnly(() => {
    commentItem(what, draftComment, item.uid, user);
})
```

you know I'm taking about ?
pm me if you need assistance sir

File: src/components/ItemView/ItemView.tsx [hunk 4]
```diff
@@ -37,6 +37,7 @@ export function Actions<T extends Photo | Post | Link | Coil | Liquid>({
 }: ItemViewProps<T>) {
   const { initialState } = useModel('@@initialState');
   const user = initialState?.currentUser as CurrentUser;
+  const currentUser = initialState?.firebaseUser as User;
```
Comment (stasbar) [inline src/components/ItemView/ItemView.tsx:40]:
```
  const { initialState } = useModel('@@initialState');
  const { currentUser, firebaseUser } = initialState || {};
```
then you need to replace all instance of `currentUser`'s to `firebaseUser`'s and `user`'s to `currentUser`'s and ``

**Review by stasbar — CHANGES_REQUESTED** [review]
Besides that, it should be good

**Review by stasbar — APPROVED** [review]

**Pull request merged**